src/injected/element-finder-by-path.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import Q from '../../node_modules/@types/q';
+import { HTMLElementUtils } from '../common/html-element-utils';
+import { ErrorMessageContent } from './frameCommunicators/error-message-content';
+import { FrameCommunicator } from './frameCommunicators/frame-communicator';
+import { FrameMessageResponseCallback } from './frameCommunicators/window-message-handler';
+
+export interface ElementFinderByPathMessage {
+    path: string[];
+}
+
+export class ElementFinderByPath {
+    public static readonly findElementByPathCommand = 'insights.findElementByPathCommand';
+
+    constructor(
+        private readonly htmlElementUtils: HTMLElementUtils,
+        private readonly frameCommunicator: FrameCommunicator,
+        private readonly q: typeof Q,
+    ) {}
+
+    public initialize(): void {
+        this.frameCommunicator.subscribe(ElementFinderByPath.findElementByPathCommand, this.onfindElementByPath);
+    }
+
+    protected onfindElementByPath = (
+        message: ElementFinderByPathMessage,
+        error: ErrorMessageContent,
+        sourceWin: Window,
+        responder?: FrameMessageResponseCallback,
+    ): void => {
+        this.processRequest(message).then(
+            result => {
+                responder(result, null, sourceWin);
+            },
+            err => {
+                responder(null, err, sourceWin);
+            },
+        );
+    };
+
+    public processRequest = (message: ElementFinderByPathMessage): Q.IPromise<string> => {
```

Comment:
I am just curious if we can avoid using `Q` in these functions? Will `Promise` API not be sufficient for your use case?

Reply:
Q officially removed! Thank you!